service-wcs/src/main/java/org/oskari/wcs/parser/CommonParser.java

```diff
@@ -0,0 +1,103 @@
+package org.oskari.wcs.parser;
+
+import java.util.Optional;
+import org.oskari.wcs.gml.Envelope;
+import org.oskari.wcs.gml.Point;
+import org.oskari.wcs.util.XML;
+import org.w3c.dom.Element;
+
+public class CommonParser {
+
+    public static Envelope parseEnvelope(Element envelope) {
+        String srsName = envelope.getAttribute("srsName");
+        String tmp = envelope.getAttribute("srsDimension");
+        if (tmp == null || tmp.isEmpty()) {
+            throw new IllegalArgumentException(
+                    "Invalid Envelope, srsDimension shall not be empty");
+        }
+        final int srsDimension = Integer.parseInt(tmp);
+
+        String[] axisLabels = null;
+        tmp = envelope.getAttribute("axisLabels");
+        if (tmp != null && !tmp.isEmpty()) {
+            axisLabels = tmp.split(" ");
+            if (axisLabels.length != srsDimension) {
+                throw new IllegalArgumentException(
+                        "Invalid Envelope, number of axisLabels differs from dimension");
+            }
+        }
+
+        String[] uomLabels = null;
+        tmp = envelope.getAttribute("uomLabels");
+        if (tmp != null && !tmp.isEmpty()) {
+            uomLabels = tmp.split(" ");
+            if (uomLabels.length != srsDimension) {
+                throw new IllegalArgumentException(
+                        "Invalid Envelope, number of uomLabels differs from dimension");
+            }
+        }
+
+        String lowerCornerStr = XML.getChildText(envelope, "lowerCorner").orElseThrow(
+                () -> new IllegalArgumentException("Could not find lowerCorner from Envelope"));
+        double[] lowerCorner = parseDoubleArray(lowerCornerStr, ' ').orElseThrow(
+                () -> new IllegalArgumentException("Invalid Envelope,"
+                        + " failed to parse doubleList from lowerCorner"));
+
+        String upperCornerStr = XML.getChildText(envelope, "upperCorner").orElseThrow(
+                () -> new IllegalArgumentException("Could not find lowerCorner from Envelope"));
+        double[] upperCorner = parseDoubleArray(upperCornerStr, ' ').orElseThrow(
+                () -> new IllegalArgumentException("Invalid Envelope,"
+                        + " failed to parse doubleList from upperCorner"));
+
+        return new Envelope(srsName, srsDimension, axisLabels, uomLabels, lowerCorner, upperCorner);
+    }
+
+    public static Optional<double[]> parseDoubleArray(String str, char c) {
```

Comment:
These parse methods (parseDoubleArray, parseIntArray, parseInt) are very similar to what ConversionHelper already has, but using Optionals. I would suggest that these should be in ConversionHelper, but that would mean a load of dependencies that you don't otherwise need.

Should we add a new utility-module that has no dependencies and could hold classes with these kinds of functions?